### ISSUE TITLE
Google Chrome 100: NVDA no longer reads the title of the newly loaded page and automatic SayAll is broken

### ISSUE THREAD

**Issue by amirsol81** [nvaccess/nvda#13571]
Google Chrome 100, and presumably MS Edge 100 which hasn't been released at the time of this issue creation, has broken some aspects of NVDA's browsing capabilities and features, resulting in the suppression of the announcement of the page title upon load, and malfunctioning of the Automatic SayAll on Page Load feature. This also tampers with the capabilities of some add-ons -- namely BrowserNav.
### Steps to reproduce:
While the issues mentioned here can be duplicated with all websites, this is just one sample selected for ease of duplication:
1. With Chrome 100, go to https://osara.reaperaccessibility.com
### Actual behavior:
NVDA just says: "Document." Moreover, if NVDA's Automatic SayAll on Page Load check box is checked, it doesn't work and NVDA remains silent. And if, I mean only if the BrowserNav add-on is enabled, moving around with Up/Down results in the second readout of many HTML elements such as links, buttons, etc., and extra error messages (not included in my attached log) will be added to the log. Pressing Control+F5 resolves the add-on issue.
### Expected behavior:
NVDA should read the title of the page upon page load. Moreover, if the relevant check box is checked, Automatic SayAll should be triggered.
#### NVDA installed/portable/running from source:
Installed, also duplicated with portable
#### NVDA version:
2021.3.5, 2022.1 Beta 2, and Alpha 25159
#### Windows version:
Windows 11 Pro 64-Bit, 21 H2, 
#### Name and version of other software in use when reproducing the issue:
Google Chrome 100 official
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
mentioned above: the latest alpha/official/beta release
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes. However, with the BrowserNav add-on, more issues will be encountered -- the add-on won't detect its bookmarks and Up/Down will add a second readout of many HTML elements unless Control+F5 is pressed to refresh the page.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tried
[Log.txt](https://github.com/nvaccess/nvda/files/8399727/Log.txt)

**Comment by PeterLecky** [comment]
This happens also with brave based on chromium 100.
Bigger pages are not loaded at all. E.g.:
Steps to reproduce:
1. open this page:
https://developer.android.com/reference/android/content/Intent
2. Start to jump by headings after short time. There will be some headings but after few more seconds, the page wil disappear and you will see only cookies confirmation block in virtual buffer.

**Comment by mazen428** [comment]
Issue is also with Edge 100.

**Comment by michaelDCurran** [comment]
It seems that in Chromium 100 there is a definite change to the events we receive when a page loads.
It looks like the following happens:
1. the user causes a page to load by pressing enter in the location bar or activating a link
2. A new document accessible is created with no name, no value, and states of focused, focusable and readonly.
3.  a focus event is fired on this document.
  3.1. NVDA receives the focus event, and tries to load a virtualBuffer for the document, but it is blank (no content) and NVDA cannot get its URL (documentConstantIdentifier) as  the document's accValue is empty still.
5. Content starts to load, and the name, value, and descendants are added.
6.  A documentLoadComplete event is fired on the document.
  6.1. Currently NVDA does not pay attention to this event, as it expects that  the first or another focus event would happen when most if not all of the document was loaded.

Previously, focus was not fired on the document until the name, value, and at least some of the descendants were loaded. And therefore, NVDA was able to load a virtualBuffer with at least some content.

I am currently looking at a fix for NVDA 2022.1 which will make use of documentLoadComplete if previous tries at loading the virtualBuffer yielded in a blank document.

This is related to #13570 
 
@aleventhal Clearly something has changed in Chromium 100, around firing of initial focus event on a new document. It looks like refreshing the document does not exhibit the issue as much if at all. Can you locate the change in Chromium? And if so, was this deliberate? As I've mentioned above, I can work around this with documentLoadComplete, and perhaps we should have always done this to begin with, however there was originally no need, and we still need to maintain compatibility with older browsers.

**Comment by aleventhal** [comment]
Thanks Mick, is there a set of easy steps to repro? Or should we look at
#13570 <https://github.com/nvaccess/nvda/issues/13570> for steps?
+Daiverd ***@***.***> , can you have the analyst team bisect it so
that we are sure we know what the exact cause was?
It's probably something from +Joanmarie Diggs ***@***.***> related
to page load events (she's back next week).

Aaron

**Comment by aleventhal** [comment]
+Chung Kim ***@***.***> who will help me bisect this issue.

I was able to repro this in these versions:

100.0.4896.75 (Official Build) (64-bit) (cohort: Stable Installs & Version
Pins)

101.0.4951.15 (Official Build) beta (64-bit) (cohort: Beta)

102.0.4987.0 (Official Build) canary (64-bit) (cohort: Clang-64)


But not in Dev:

102.0.4972.0 (Official Build) dev (64-bit) (cohort: Dev)


Chung pointed out that whatever feature is causing this issue may not be
included in the dev build.


We will update with bisect results.


Thanks,
Jen

**Comment by joanmarie** [comment]
Sorry for just now getting to this. I was not working last week.

I went ahead and did the bisect (using `bisect-builds.py`). The [list of commits](https://chromium.googlesource.com/chromium/src/+log/a65f2021c068ce6c56530cbada3d6633c7950825..1ec123d476397a5773946c5e4d5bfc8f5d737bf4) is fairly large, but [my change related to firing load-start](https://chromium.googlesource.com/chromium/src/+/fca4bbc9a10ff2d5b2ba31f81f817a12af260bdc) is in there and the likeliest suspect.

I see that @michaelDCurran has landed a fix on the NVDA side of things. So should I still try to adjust things in Chromium to get things working as expected without the need for NVDA users to upgrade?

**Comment by joanmarie** [comment]
okie dokie. I've opened [this bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1315210) and will see how we can fix it on the Chromium side.

**Comment by aleventhal** [comment]
Mick, is my understanding correct? We can't just slipstream this into
current versions or something.

**Comment by michaelDCurran** [comment]
You are correct that users are very slow to upgrade.

However, there is nothing stopping them doing so... only their choice 
really.

We have a fix for NVDA 2022.1 which will be released very soon. So its 
up to Google as to whether this issue should be fixed.

If it was not deliberate, then I'd push for it to be fixed, as anything 
not deliberate may be unpredictable in future. But if it was deliberate, 
we have code that can work with it now.

Mick

**Comment by aleventhal** [comment]
From Brian's bisect. "You are probably looking for a change made after
965887 (known good), but no later than 965914 (first known bad).
CHANGELOG URL:

https://chromium.googlesource.com/chromium/src/+log/43dcd996d18f50ef0a4903c6b69b59413d730d2a..f117e945c461e303edbd8d6f47439c74b8b5fbf4
"

**Comment by joanmarie** [comment]
> Hey, do we have a bisect for this and/or do we have a crbug link?

Did you see https://github.com/nvaccess/nvda/issues/13571#issuecomment-1095032095?

**Comment by aleventhal** [comment]
Sorry, I'd been responding in the github issue rather than this thread.
I think that bisect is incorrect. Here's what I
got: https://chromium.googlesource.com/chromium/src/+log/a65f2021c068ce6c56530cbada3d6633c7950825..1ec123d476397a5773946c5e4d5bfc8f5d737bf4

I believe my change is indeed to blame and plan to find a solution (I
have some ideas).

Here is the
bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1315210

**Comment by amirsol81** [comment]
@michaelDCurran @aleventhal It seems that the commit provided by NV Access hasn't entirely fixed the issue with the readout of newly loaded pages. For instance, go to https://www.npr.org/programs/morning-edition/archive and select whatever link which is embedded in level-3 headings -- links like "In Paris, an art collector paid $1.2 million for a receipt" or "A stripped-down 'Cyrano' — a London import — is all about language." As we select these links and a new page loads, all I hear is the phrase "visited link," without any indication regarding the page title. Refreshing the page with CTRL+F5 forces NVDA to read the title, though.
I'm using NVDA Alpha 25209, with or without add-ons, and Google Chrome 100 official.
